### PR TITLE
Added endpoints for each language and integration with languages

### LINKED ISSUE #12: Spring API Endpoints
- [x] Individual endpoints for each language.
- [x] Create a file.

### LINKED ISSUE #26: Save the file created into a local host

### PR DESCRIPTION
**Pull request by nanrehd**

### CODE REVIEW

**Review by carlosmc23 — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -14,6 +14,7 @@ repositories {
 }
 
 dependencies {
+    compile group: 'com.googlecode.json-simple', name: 'json-simple', version: '1.1'
     testCompile 'org.junit.jupiter:junit-jupiter-api:5.6.0'
```
Comment (luqp) [inline build.gradle:18]: you can use the last version here (1.1.1)

File: src/main/java/org/jalasoft/moi/controller/HelloController.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
     /**
      * @return a constraint string for a simple example
      */
-    @GetMapping
+    @GetMapping("/hello")
     public String hello() {
         return "Hello!!";
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/controller/HelloController.java]: I think this class is not necessary.

File: src/main/java/org/jalasoft/moi/controller/cplusplus/CppController.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @param jsonRequest A String representing a JSON.
+     * @return the output from the execution.
+     */
+    @RequestMapping(path = "/v1/onlineCompiler/cpp", method = RequestMethod.POST, consumes = "application/json")
+    public String executeSingleCode(@RequestBody String jsonRequest) throws ParseException, IOException {
+        codeParams = codeHelper.codeParams(jsonRequest);
+        String result = codeHelper.execCode(codeParams);
+        return result;
+    }
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/controller/cplusplus/CppController.java:48]: You can simplify the last 2 lines in one like this: `return codeHelper.execCode(codeParams);`

File: src/main/java/org/jalasoft/moi/controller/csharp/CSharpController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+     *
+     * @param jsonRequest A String representing a JSON.
+     * @return the output from the execution.
+     */
+    @RequestMapping(path = "/v1/onlineCompiler/csharp", method = RequestMethod.POST, consumes = "application/json")
+    public String executeSingleCode(@RequestBody String jsonRequest) throws ParseException, IOException {
+        codeParams = csharpHandler.codeParams(jsonRequest);
+        String result = csharpHandler.execute(codeParams);
+        return result;
+    }
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/controller/csharp/CSharpController.java:47]: Here a similar case.

File: src/main/java/org/jalasoft/moi/controller/python/PythonController.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.IOException;
+
+/**
+ * This class defines the controller for Python.
+ *
+ * @version        1.0
+ * @author         Diego Perez
+ */
+@RestController("/py")
+public class PythonController {
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/controller/python/PythonController.java:32]: If you are put complete names for all languages this should follow the same convention:
use -> `/python`
instead -> `/py`

File: src/main/java/org/jalasoft/moi/model/core/IHandler.java [hunk 6]
```diff
@@ -22,4 +27,15 @@
      * @return a String of result from CommandBuilder and Executer handling
      */
     String execute(Params params);
+
+    /**
+     * Returns a Params object.
+     * A JSON object gets deconstructed and its data used to make a
+     * Params object.
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/model/core/IHandler.java:35]: For conventions here should be an space.

File: src/main/java/org/jalasoft/moi/model/core/IHandler.java [hunk 7]
```diff
@@ -22,4 +27,15 @@
...
+     * @param jsonRequest A string ready to be decomposed into variables.
+     * @return A Params object.
+     * @throws IOException
+     * @throws ParseException
+     */
+    Params codeParams(String jsonRequest) throws IOException, ParseException;
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/model/core/IHandler.java]: I believe that it should have a more significant name
I suggest: `convertToParams(...)`

File: src/main/java/org/jalasoft/moi/model/csharp/CsharpHandler.java [hunk 8]
```diff
@@ -44,4 +51,32 @@ public String execute(Params params) {
...
+        String language = (String) jsonObject.get("language");
+        String version = (String) jsonObject.get("version");
+        //Creates and writes a file with the code needed.
+        File codeFile = new File("C:\\Users\\Admin\\Documents\\temp\\" + fileName);
+        FileWriter codeWriter = new FileWriter("C:\\Users\\Admin\\Documents\\temp\\" + fileName);
+        codeWriter.write(code);
+        codeWriter.close();
+        Params codeParams = new Params(codeFile.toPath(),
+                                        version,
+                                        Language.CSHARP);
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/model/csharp/CsharpHandler.java]: the Params instantiation could be an in a single line

File: build.gradle (hunk 1 above)
Comment (nanrehd) [reply build.gradle:18]: Will do, thanks!!

File: src/main/java/org/jalasoft/moi/controller/HelloController.java (hunk 2 above)
Comment (nanrehd) [reply src/main/java/org/jalasoft/moi/controller/HelloController.java]: Example classes will be deleted soon.

**Review by luqp — DISMISSED** [review]

**Review by carlosmc23 — DISMISSED** [review]

**Review by luqp — APPROVED** [review]

**Review by carlosmc23 — APPROVED** [review]

**Pull request merged**